### DISCUSSION TITLE
Export results to sklearn/tensorflow/xgboost/catboost class

### DISCUSSION THREAD

**Discussion by gkositsk** [autogluon/autogluon#2126] [Q&A] [answered]
Hi,
Is there a way to train TabularPredictor, and then get the best models as class instances of the origin libraries?(sklearn/xgboost/catboost/tensorflow).
I read the docs and couldn't find any direct way to do it.

I'll be glad to get help with that.

**Comment by easyautoml** [comment, marked as answer]
Hi @gkositsk, used bellow code to load origin model. 
```
model = predictor._trainer.load_model(model_name).model
```

**Reply by gkositsk** [reply]
Thank you

**Reply by gkositsk** [reply]
Hi @easyautoml , I tried it again after a while.. when acssesing this property I always get None. Is there a way to fix it?

**Reply by easyautoml** [reply]
Sorry response you late. The original model can only be loaded if it is Tree Model such as Random forest, Catboost or XGBoost. Tree Models is builded by using Sklearn while Neural Network models are different. 

You can figure out the reason when compare the source code of Random Forest and Neural Network at bellow url. 
Otherwise, you'd better spend more effort to read bellow source code if you want to use the function that are not officially support by Autogluon.
 
https://github.com/autogluon/autogluon/tree/master/tabular/src/autogluon/tabular/models

**Comment by FailedPipeline** [comment]
@gkositsk how did you then use this in the origin libraries?

**Reply by gkositsk** [reply]
Not in a special way, I needed to store the models as pickle files, and later use them. right now the process expects objects from these libraries.
